### PR TITLE
Create new top level directory for package  related clients

### LINKED ISSUE #3287: Extract package clients from TKG to top level packageclients directory as a separate go module
**Describe the feature request**
Restructure the clients needed by both package/secret plugins.
Create a top level directory with its own go module to facilitate subsequent move of it to its own repository.

There are 3 go packages ```tkgpackageclient```, ```kappclient```, ```tkgpackagedatamodel```, that need to be separated out into its own top level directory with a dedicated go module.
secret plugin, package plugin and management cluster plugin will import this as a dependency.
move integration tests to the cmd layer and update github workflows and test-infra code.

**Describe alternative(s) you've considered**

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- (X) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- (X) Plugin
- ( ) Security
- (X) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by maralavi**
### What this PR does / why we need it
Create new top level directory for package related clients

### Which issue(s) this PR fixes

Fixes #3287 

### Release note

```release-note
None
```

### Additional information

Moved the following directories from pkg/v1/tkg to packageclients/ module:

- tkgpackageclient
- tkgpackagedatamodel
- kappclient

### CODE REVIEW

**Review by anujc25 — APPROVED** [review]
Thanks for the changes. This looks good to me.

cc @vuil @tenczar @vijaykatam

File: cmd/cli/plugin/package/test/package_plugin_integration_test.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2021 VMware, Inc. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package tkgpackageclient_test
+package main
```
Comment (vijaykatam) [inline cmd/cli/plugin/package/test/package_plugin_integration_test.go:4]: Question: Why is the test package renamed to main?

File: packageclients/pkg/fakes/crtclient.go [hunk 2]
```diff
@@ -0,0 +1,776 @@
+// Code generated by counterfeiter. DO NOT EDIT.
```
Comment (vijaykatam) [inline packageclients/pkg/fakes/crtclient.go:1]: Question: Where are code gen markers for these fakes located? We will also need a make target within packageclients `Makefile` that generates them.

File: pkg/v1/tkr/Dockerfile [hunk 3]
```diff
@@ -15,6 +15,7 @@ COPY go.mod go.mod
 COPY go.sum go.sum
 COPY apis/ apis/
 COPY cli/runtime cli/runtime
+COPY packageclients/ packageclients/
```
Comment (vijaykatam) [inline pkg/v1/tkr/Dockerfile:18]: We discussed offline that is an interim step required for tkr components to be able to build. Should go away when tkr components are refactored to their own go mod.

**Review by vijaykatam — COMMENTED** [review]
Change looks good for the most part. We are missing Makefile for packageclients go module that should at a minimum do vet, fmt, test and code gen for fakes.

File: cmd/cli/plugin/package/test/package_plugin_integration_test.go (hunk 1 above)
Comment (maralavi) [reply cmd/cli/plugin/package/test/package_plugin_integration_test.go:4]: Cause I moved it to the ```main``` package of ```cmd/cli/plugin/package/test``` from ```pkg/v1/tkg/test/packageclients``` and the package name needed to match  ```cmd/cli/plugin/package/test``` and wouldn't accept multiple package names.

File: packageclients/pkg/fakes/crtclient.go (hunk 2 above)
Comment (maralavi) [reply packageclients/pkg/fakes/crtclient.go:1]: > Question: Where are code gen markers for these fakes located?

It is in the interface definitions for each client

Comment (maralavi) [reply packageclients/pkg/fakes/crtclient.go:1]: > We will also need a make target within packageclients Makefile that generates them.
done.

File: packageclients/pkg/packageclient/image_utils.go [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2021 VMware, Inc. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package tkgpackageclient
+package packageclient
```
Comment (codegold79) [inline packageclients/pkg/packageclient/image_utils.go:4]: Usually, package names are not in camel case. Also, to be consistent with others, shouldn't this be `packageclient`?

File: packageclients/pkg/packageclient/interface.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
```
Comment (codegold79) [inline packageclients/pkg/packageclient/interface.go:1]: Should this be 2022?

**Review by codegold79 — CHANGES_REQUESTED** [review]
Great work @maralavi. Only a couple very minor comments.

File: packageclients/pkg/packageclient/interface.go (hunk 5 above)
Comment (maralavi) [reply packageclients/pkg/packageclient/interface.go:1]: I had originally written this file in 2021, I just moved it to a new directory. So I think date remains as it was.

File: packageclients/pkg/packageclient/image_utils.go (hunk 4 above)
Comment (maralavi) [reply packageclients/pkg/packageclient/image_utils.go:4]: This is not camelcase. And I've already renamed it to ```packageclient```, so it is correct.

**Review by codegold79 — APPROVED** [review]

File: packageclients/pkg/packageclient/image_utils.go (hunk 4 above)
Comment (codegold79) [reply packageclients/pkg/packageclient/image_utils.go:4]: I might have been seeing the old version. When I left the review comment, I saw the package name as `packageClient` (camel case). But I see now it is how I had suggested, `packageclient`.

**Review by vijaykatam — APPROVED** [review]

**Pull request merged**